fix(napi/parser): remove non-existent methods from TS type defs

overlookmotel opened the pull request:
Fix #11950.

Remove `parseSyncRaw`, `parseAsyncRaw`, and `getBufferOffset` from TS type defs. They're not actually exported from the package.

Boshen commented:
### Merge activity

* **Jul 3, 1:04 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Jul 3, 1:05 PM UTC**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
fix(napi/parser): remove non-existent methods from TS type defs (#12054)

Fix #11950.

Remove `parseSyncRaw`, `parseAsyncRaw`, and `getBufferOffset` from TS type defs. They're not actually exported from the package.
